vue/components/ui/molecules/scrollable-items/scrollable-items.vue

```diff
@@ -0,0 +1,260 @@
+         */
+        alignment: {
+            type: String,
+            default: "center"
+        },
+        /**
+         * Dictates the horizontal scroll speed of the items when using the mouse
+         * scroll wheel. Negative values inverts the default direction of the scroll.
+         */
+        scrollSpeed: {
+            type: Number,
+            default: 10
+        },
+        /**
+         * Props for the arrow button icon components.
+         */
+        buttonIconProps: {
+            type: Object,
+            default: () => ({})
+        }
+    },
+    data: function() {
+        return {
+            selectedData: this.selected,
+            isMouseDown: false,
+            isDraggingItems: false,
+            scrollableWidth: 0
+        };
+    },
+    computed: {
+        itemsContainerStyle() {
+            return {
+                "text-align": this.alignment
+            };
+        },
+        _buttonIconProps() {
+            return {
+                borderRadius: 0,
+                iconStrokeWidth: 3,
+                ...this.buttonIconProps
+            };
+        },
+        showArrows() {
+            const hideArrows = this.autoHideArrows && !(this.scrollableWidth > 0);
+            return this.arrows && !hideArrows;
+        },
+        selectedIndex() {
+            return this.items.findIndex(item => item.value === this.selectedData);
+        },
+        isButtonPreviousDisabled() {
+            return this.selectedIndex <= 0;
+        },
+        isButtonNextDisabled() {
+            return this.selectedIndex >= this.items.length - 1;
+        }
+    },
+    watch: {
+        selected(value) {
+            this.selectedData = value;
+        },
+        selectedData(value) {
+            this.snapSelectedToCenter();
+            this.$emit("update:selected", value);
+        }
+    },
+    created: function() {
+        window.addEventListener("resize", this.onResize);
+        window.addEventListener("mouseup", this.onMouseUp);
+        window.addEventListener("mousemove", this.onMouseMove);
+    },
+    destroyed: function() {
+        window.removeEventListener("mousemove", this.onMouseMove);
+        window.removeEventListener("mouseup", this.onMouseUp);
+        window.removeEventListener("resize", this.onResize);
+    },
+    methods: {
+        itemClasses(item, index) {
+            const base = {};
+            base[`item-${item.value}`] = true;
+            if (item.value === this.selectedData) base.selected = true;
+            if (this.items.length && index === this.items.length - 1) base.last = true;
+            return base;
+        },
+        calculateScrollableWidth() {
+            if (!this.$refs["items-container"]) return;
+            this.scrollableWidth = this.$refs["items-container"].scrollLeftMax;
+        },
+        snapSelectedToCenter() {
+            this.$refs["items-container"].childNodes[this.selectedIndex].scrollIntoView({

```

Comment:
I believe this doesn't work in Safari, but give it a try.

Reply:
If it doesn't this will have to be worked around with a different implementation.

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior might help